bigtable_rs/src/bigtable.rs

```diff
@@ -329,29 +328,47 @@ impl BigTableConnection {
         project_id: &str,
         instance_name: &str,
         is_read_only: bool,
-        channel_size: usize,
         timeout: Option<Duration>,
     ) -> Result<Self> {
         info!("Connecting to bigtable emulator at {}", emulator_endpoint);
-        let endpoints: Vec<Endpoint> = vec![0; channel_size.max(1)]
-            .iter()
-            .map(move |_| {
-                Channel::from_shared(format!("http://{}", emulator_endpoint))
-                    .expect("Invalid connection emulator uri")
-                    .http2_keep_alive_interval(Duration::from_secs(60))
-                    .keep_alive_while_idle(true)
-            })
-            .map(|ep| {
-                if let Some(timeout) = timeout {
-                    ep.timeout(timeout)
-                } else {
-                    ep
-                }
-            })
-            .collect();
+
+        // Parse emulator_endpoint. Officially, it's only host:port,
+        // but unix:///path/to/unix.sock also works in the Go SDK at least.
+        // Having the emulator listen on unix domain sockets without ip2unix is
+        // covered in https://github.com/googleapis/google-cloud-go/pull/9665.
+
+        let endpoint = if emulator_endpoint.starts_with("unix://") {
+            // the URL doesn't matter, we use a custom connector later.
+            Endpoint::from_static("http://[::]:50051")
+        } else {
+            Channel::from_shared(format!("http://{}", emulator_endpoint))
+                .expect("invalid connection emulator uri")
+        };
+
+        let endpoint = endpoint
+            .http2_keep_alive_interval(Duration::from_secs(60))
+            .keep_alive_while_idle(true);
+        let endpoint = if let Some(timeout) = timeout {
+            endpoint.timeout(timeout)
+        } else {
+            endpoint
+        };
+
+        let channel = if let Some(path) = emulator_endpoint.strip_prefix("unix://") {
+            let uri = tonic::transport::Uri::from_maybe_shared(path.to_owned())
+                .expect("invalid connection emulator uri");
+            let connector = tower::service_fn({
+                let uri = uri.clone();
+                move |_: tonic::transport::Uri| UnixStream::connect(uri.path().to_string().clone())
+            });
+
+            endpoint.connect_with_connector_lazy(connector)
```

Comment:
happy to also use the non-lazy counterparts (and make `new_with_emulator` async too), just let me know :-)

Reply:
I started using the non-lazy counterparts in the current iteration of this PR.